src/main/java/org/ibp/api/java/impl/middleware/role/RoleServiceImpl.java

```diff
@@ -27,4 +42,60 @@
 
 		return roles;
 	}
+
+	@Override
+	public void createRole(final RoleGeneratorInput dto) {
+		//TODO Check by SITE_ADMIN authority
+
+		this.roleValidator.validateRoleGeneratorInput(dto);
+
+		final String userName = SecurityUtil.getLoggedInUserName();
+		final WorkbenchUser user = this.workbenchDataManager.getUserByUsername(userName);
+		final Role role  = new Role();
+		role.setName(dto.getName());
+		role.setEditable(dto.isEditable());
+		role.setActive(dto.isActive());
+		role.setAssignable(dto.isAssignable());
+		role.setDescription(dto.getDescription());
+		role.setCreatedDate(new Date());
+		role.setCreatedBy(user);
+		role.setActive(dto.isActive());
+		role.setPermissions(this.getPermission(dto.getPermissions()));
+		role.setRoleType(this.getRoleType(dto.getRoleType()));
+		role.setUpdatedBy(user);
+		role.setUpdatedDate(new Date());
+		this.workbenchDataManager.saveOrUpdateRole(role);
+	}
+
+	private RoleType getRoleType(final Integer roleTypeId) {
+		final RoleType roleType =  this.workbenchDataManager.getRoleType(roleTypeId);
+		return roleType;
+	}
+
+	private List<Permission> getPermission(final List<Integer> permissions) {
+
+		final List<Permission> permissionList = new ArrayList<>();
+		for (final Integer permissionId : permissions) {
+			final Permission permission = this.workbenchDataManager.getPermission(permissionId);
+			permissionList.add(permission);
+
+		}
+
+		final Collection result = CollectionUtils.collect(permissionList, new Transformer() {
```

Comment:
@vmaletta is it this list content equals to permissions, given that you have already validated that all the ids exists?

Reply:
No, I didn't

Reply:
You did it in the validator